test/Kestrel.Core.Tests/KestrelServerTests.cs

```diff
@@ -327,6 +327,63 @@ public void StartWithNoTransportFactoryThrows()
             mockTransport.Verify(transport => transport.UnbindAsync(), Times.Once);
         }
 
+        [Fact]
+        public async Task StopAsyncDispatchesSubsequentStopAsyncContinuations()
+        {
+            var options = new KestrelServerOptions
+            {
+                ListenOptions =
+                {
+                    new ListenOptions(new IPEndPoint(IPAddress.Loopback, 0))
+                }
+            };
+
+            var unbindTcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            var mockTransport = new Mock<ITransport>();
+            mockTransport
+                .Setup(transport => transport.BindAsync())
+                .Returns(Task.CompletedTask);
+            mockTransport
+                .Setup(transport => transport.UnbindAsync())
+                .Returns(unbindTcs.Task);
+            mockTransport
+                .Setup(transport => transport.StopAsync())
+                .Returns(Task.CompletedTask);
+
+            var mockTransportFactory = new Mock<ITransportFactory>();
+            mockTransportFactory
+                .Setup(transportFactory => transportFactory.Create(It.IsAny<IEndPointInformation>(), It.IsAny<IConnectionDispatcher>()))
+                .Returns(mockTransport.Object);
+
+            var mockLoggerFactory = new Mock<ILoggerFactory>();
```

Comment:
Why can't we use this.LoggerFactory?

Reply:
KestrelServerTests is not a TestApplicationErrorLoggerLoggedTest.